### PR TITLE
Dev/rebuild plugin system

### PR DESCRIPTION
**Pull request by binyamin555**

### CODE REVIEW

File: modules/plugins/builtins/loader.py
Comment (neriyaco) [inline modules/plugins/builtins/loader.py:1]: Finally

File: core/ez/__init__.py [hunk 1]
```diff
@@ -389,7 +388,19 @@ def _setup(__ez=_EZ.ez):
...
     from ez.events import Modules
+    from plugins.ez_plugins import Plugins, __pm
+
+    @once(Plugins.WillLoad)
+    def _(_):
+        type(__ez).currently_loaded_plugin = property(lambda _: __pm.current_plugin)
+
+    @once(Plugins.DidLoad)
+    def _(_):
+        type(__ez).currently_loaded_plugin = property(lambda _: None)
```
Comment (neriyaco) [inline core/ez/__init__.py:403]: WTF

File: core/ez/__init__.py [hunk 2]
```diff
@@ -400,19 +411,7 @@ def _setup(__ez=_EZ.ez):
 
     del Modules
 
-    from ez.plugins import PluginEvent, __pm
-
-    plugins = ["test-plugin", "title-changer"]
```
Comment (neriyaco) [inline core/ez/__init__.py:405]: Thank god

**Review by neriyaco — APPROVED** [review]

File: core/ez/__init__.py (hunk 1 above)
Comment (binyamin555) [reply core/ez/__init__.py:403]: Yeah, ik

File: modules/plugins/builtins/loader.py
Comment (binyamin555) [reply modules/plugins/builtins/loader.py:1]: Water you talking about exactly

File: core/ez/__init__.py (hunk 2 above)
Comment (binyamin555) [reply core/ez/__init__.py:405]: Indeed

**Pull request merged**